lib/views/question/question_view.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:kayla_flutter_ic/gen/assets.gen.dart';
+import 'package:kayla_flutter_ic/utils/app_bar_ext.dart';
+import 'package:kayla_flutter_ic/views/question/question_container_view_model.dart';
+
+class QuestionView extends ConsumerStatefulWidget {
+  final Widget child;
+  final Function() onNextQuestion;
+  final Function() onSubmit;
+
+  const QuestionView({
+    super.key,
+    required this.child,
+    required this.onNextQuestion,
+    required this.onSubmit,
+  });
+
+  @override
+  QuestionViewState createState() => QuestionViewState();
+}
+
+class QuestionViewState extends ConsumerState<QuestionView> {
+  AppBar get _appBar => AppBarExt.appBarWithCloseButton(context: context);
+
+  Widget get _background => SizedBox(
+        width: MediaQuery.of(context).size.width,
+        height: MediaQuery.of(context).size.height,
+        // TODO: - Bind question's photo
+        child: Image(image: Assets.images.nimbleBackground.image().image),
+      );
+
+  Widget get _mainBody => Container(
+        width: MediaQuery.of(context).size.width,
+        padding: const EdgeInsets.all(20),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            _questionNumber,
+            const SizedBox(
+              height: 16,
+            ),
+            _questionTitle,
+            SingleChildScrollView(child: widget.child),
+          ],
+        ),
+      );
+
+  Widget get _questionNumber => Consumer(builder: (_, ref, __) {
+        final uiModel = ref.watch(questionContainerUiModelStream).value;
```

Comment:
we should put `uiModel` to the parameter list of `QuestionView` so we don't need to use Consumer. The idea is to make `QuestionView` as a `StatelessWidget` -> Increase render performance 😁